Sources/Operators/RetryWhen.swift

```diff
@@ -0,0 +1,102 @@
+//
+//  RetryWhen.swift
+//  CombineExt
+//
+//  Created by Daniel Tartaglia on 3/21/20.
+//
+
+#if canImport(Combine)
+import Combine
+
+@available(OSX 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *)
+public extension Publisher {
+
+    /// Repeats the source publisher on error when the notifier emits a next value. If the source publisher errors and the notifier completes, it will complete the source sequence.
+    ///
+    /// - Parameter notificationHandler: A handler that is passed a publisher of errors raised by the source publisher and returns a publisher that either continues, completes or errors. This behavior is then applied to the source publisher.
+    /// - Returns: A publisher producing the elements of the given sequence repeatedly until it terminates successfully or is notified to error or complete.
+    func retryWhen<Trigger>(_ notificationHandler: @escaping (AnyPublisher<Self.Failure, Never>) -> Trigger)
```

Comment:
notificationHandler feels like a strange name, do you have other thoughts for naming here?

Reply:
If I keep it as a noun phrase, I could see changing it into `errorStream`? As a verb phrase, maybe `bindError[Stream]`?

Reply:
Can't it be just `trigger` ? By the way doesn't retryWhen usually accept the error itself? Or it always gets a publisher? I don't remember the RxSwifrt variant.

Reply:
Sure, trigger is fine. The RxSwift variant receives an `Observable<Error>`.

Reply:
Wait, I use `Trigger` as the output of the operator... Should I change that if the input is being named `trigger`?

Reply:
No, I think it's fine :) You accept an external trigger and output your own trigger, I don't have an issue with it. You can name the generic RetryTrigger if you really want to.

Reply:
The two `Trigger`s did cause a compile issue, so I named the external one `RetryTrigger` and the internal one `ErrorTrigger`...